File: 283_move_zeros/solution_1/move_zeros.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n_zeros = 0
        res = []
        for elem in nums:
            if elem == 0:
                n_zeros += 1
                continue
            
            res.append(elem)

        for _ in range(n_zeros):
            res.append(0)

        nums.clear()
        for elem in res:
            nums.append(elem)

File: 283_move_zeros/solution_1/move_zeros.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        vector<int> res;
        int n_zeros = 0;
        for(auto elem : nums) {
            if(elem == 0)
            {
                n_zeros++;
                continue;
            }
            res.push_back(elem);
        }
        for(; n_zeros > 0; n_zeros--)
            res.push_back(0);

        nums = res;
    }
};
